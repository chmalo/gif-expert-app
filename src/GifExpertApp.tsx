import { useState } from "react"
import AddCategory from "./components/AddCategory"
import GifGrid from "./components/GifGrid"

const GifExpertApp = ({
  defaultCategory = [],
}: {
  defaultCategory?: string[]
}) => {
  // const [categories, setCategories] = useState(["Rick and Morty"])
  const [categories, setCategories] = useState(defaultCategory)

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category: string) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  )
}

export default GifExpertApp
