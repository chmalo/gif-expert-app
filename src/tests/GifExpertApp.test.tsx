import { shallow } from "enzyme"
import { describe, expect, test } from "vitest"
import GifExpertApp from "../GifExpertApp"

describe("testing GifExpertApp", () => {
  test("it should show correctly", () => {
    const wrapper = shallow(<GifExpertApp />)

    expect(wrapper).toMatchSnapshot()
  })

  test("it should show a categories list", () => {
    const categories = ["One Punch", "Dragon Ball"]
    const wrapper = shallow(<GifExpertApp defaultCategory={categories} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find("GifGrid").length).toBe(categories.length)
  })
})
