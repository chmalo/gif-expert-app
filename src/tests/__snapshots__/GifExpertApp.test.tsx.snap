// Vitest Snapshot v1

exports[`testing GifExpertApp > it should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`testing GifExpertApp > it should show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
